middleware/middleware_test.go

```diff
@@ -73,3 +73,32 @@ func TestMaxConcurrentRequestsMiddleware(t *testing.T) {
 		})
 	}
 }
+
+func TestContentTypeMiddleware(t *testing.T) {
+	nextHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
We could move `nextHandler` inside each test case and assert that our new middleware actually calls it.

Example:

```go
	t.Run("test SetJsonContentType middleware", func(t *testing.T) {
		called := make(chan struct{})
		nextHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
			w.WriteHeader(http.StatusOK)
			close(called)
		})
		expectedContentType := "application/json; charset=utf-8"
		contentTypeMiddleware := middleware.SetJsonContentType()
		handlerToTest := contentTypeMiddleware(nextHandler)
		respRecorder := httptest.NewRecorder()
		req := httptest.NewRequest("GET", "http://testing", nil)

		handlerToTest.ServeHTTP(respRecorder, req)
		receivedContentType := respRecorder.Header()["Content-Type"][0]
		require.Equal(t, expectedContentType, receivedContentType, "actual content type different than expected.")
		<-called // could do a select with a timeout of 1s here and fail if not called within 1s
	})
```

Reply:
I'm saying this because if you comment out line 80 (i.e. `w.WriteHeader(http.StatusOK)`) the test passes, so I think we could have a stronger guarantee that next is called.

The tests pass also if you comment out `next.ServeHTTP(w, r)` inside the middlewares.